File: (2022)/March/calculadora_financiera_java/CalculadoraFinanciera.java
// Gustavo Cocone Zavaleta | GitHub: @EgirlBuchona | © 2022
// Proyecto: Calculadora Financiera en Java

import java.util.Scanner;

public class CalculadoraFinanciera {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Calculadora Financiera");
        System.out.println("1. Cálculo de Interés Compuesto");
        System.out.println("2. Cálculo de Pago de Préstamo");
        System.out.print("Seleccione una opción: ");
        int opcion = scanner.nextInt();

        switch (opcion) {
            case 1:
                calcularInteresCompuesto(scanner);
                break;
            case 2:
                calcularPagoPrestamo(scanner);
                break;
            default:
                System.out.println("Opción no válida.");
        }
        scanner.close();
    }

    private static void calcularInteresCompuesto(Scanner scanner) {
        System.out.print("Ingrese el capital inicial: ");
        double capital = scanner.nextDouble();
        System.out.print("Ingrese la tasa de interés anual (en %): ");
        double tasa = scanner.nextDouble() / 100;
        System.out.print("Ingrese el número de años: ");
        int años = scanner.nextInt();

        double montoFinal = capital * Math.pow(1 + tasa, años);
        System.out.printf("El monto final después de %d años es: %.2f%n", años, montoFinal);
    }

    private static void calcularPagoPrestamo(Scanner scanner) {
        System.out.print("Ingrese el monto del préstamo: ");
        double prestamo = scanner.nextDouble();
        System.out.print("Ingrese la tasa de interés anual (en %): ");
        double tasa = scanner.nextDouble() / 100 / 12;  // tasa mensual
        System.out.print("Ingrese el número de pagos: ");
        int pagos = scanner.nextInt();

        double pagoMensual = (prestamo * tasa) / (1 - Math.pow(1 + tasa, -pagos));
        System.out.printf("El pago mensual es: %.2f%n", pagoMensual);
    }
}

File: (2022)/May/calculadora_financiera_python/calculadora_financiera.py
# Gustavo Cocone Zavaleta | GitHub: @EgirlBuchona | © 2022
# Proyecto: Calculadora Financiera en Python

def calcular_interes_simple(capital, tasa, tiempo):
    return capital * (tasa / 100) * tiempo

def calcular_interes_compuesto(capital, tasa, tiempo):
    return capital * ((1 + (tasa / 100)) ** tiempo - 1)

def calcular_pago_prestamo(principal, tasa, plazo):
    tasa_mensual = tasa / (12 * 100)
    return principal * tasa_mensual / (1 - (1 + tasa_mensual) ** -plazo)

if __name__ == "__main__":
    while True:
        print("\nCalculadora Financiera")
        print("1. Cálculo de Interés Simple")
        print("2. Cálculo de Interés Compuesto")
        print("3. Cálculo de Pago de Préstamo")
        print("4. Salir")
        
        opcion = input("Seleccione una opción: ")
        
        if opcion == "1":
            capital = float(input("Ingrese el capital: "))
            tasa = float(input("Ingrese la tasa de interés (%): "))
            tiempo = float(input("Ingrese el tiempo en años: "))
            print(f"Interés Simple: {calcular_interes_simple(capital, tasa, tiempo):.2f}")
        
        elif opcion == "2":
            capital = float(input("Ingrese el capital: "))
            tasa = float(input("Ingrese la tasa de interés (%): "))
            tiempo = float(input("Ingrese el tiempo en años: "))
            print(f"Interés Compuesto: {calcular_interes_compuesto(capital, tasa, tiempo):.2f}")
        
        elif opcion == "3":
            principal = float(input("Ingrese el monto del préstamo: "))
            tasa = float(input("Ingrese la tasa de interés anual (%): "))
            plazo = int(input("Ingrese el plazo en meses: "))
            print(f"Pago mensual del préstamo: {calcular_pago_prestamo(principal, tasa, plazo):.2f}")
        
        elif opcion == "4":
            print("Saliendo de la calculadora financiera.")
            break
        else:
            print("Opción no válida. Intente de nuevo.")
